res/controllers/Numark-Mixtrack-3-scripts.js

```diff
@@ -2067,157 +1754,109 @@ NumarkMixtrack3.OnPADLoopButtonChange = function(value, group, control) {
     
     // on initialization of deck, the value "0" would cause the pad LED to stop blinking on the 2nd pass
     // of triggers. This gives ensures that the PAD mode reflect the proper state of AutoLoop
-    for (i = 0; i < loopsize.length; i++) {
-        index = i + 1;
+    for (var i = 0; i < loopsize.length; i++) {
+        var index = i + 1;
+
         if (index > 4) {
             index = index - 4;
         }
 
-        var test = engine.getValue("[Channel" + decknum + "]", "beatloop_" + loopsize[i] +
-                "_enabled");
+        var test = engine.getValue(deck.group, "beatloop_" + loopsize[i] + "_enabled");
 
         if (test) {
-            NumarkMixtrack3.decks["D" + decknum].LEDs["PADloop" + index].flashOn(300, PADcolors.yellow, 300);
+            deck.LEDs["PADloop" + index].flashOn(300, PADcolors.yellow, 300);
         }
     }
 };
 
-NumarkMixtrack3.StripTouchEffect = function(channel, control, value, status,
-    group) {
-    var decknum = NumarkMixtrack3.deckFromGroup(group);
-    var deck = NumarkMixtrack3.decks["D" + decknum];
+NumarkMixtrack3.StripTouchEffect = function(channel, control, value, status, group) {
+    var deck = NumarkMixtrack3.deckFromGroup(group);
+    var focusedEffect = deck.getFocusedEffect();
+
     if (deck.shiftKey) {
-        engine.setValue("[Channel" + decknum + "]", "playposition", value / 127);
+        engine.setValue(deck.group, "playposition", value / 127);
     } else {
-        deck.StripEffect(value, decknum);
+        if (deck.InstantFX.length) {
+            for (var i = 0; i < deck.InstantFX.length; i++) {
+                var fxGroup = "[EffectRack1_EffectUnit" + deck.decknum + "_Effect" + deck.InstantFX[i] + "]";
+                engine.setValue(fxGroup, "enabled", true);
+                engine.setValue(fxGroup, "meta", value / 127);
+            }
+        }
+
+        if (focusedEffect) {
+            var focusedGroup = "[EffectRack1_EffectUnit" + deck.decknum + "_Effect" + focusedEffect + "]";
+            engine.setValue(focusedGroup, "meta", value / 127);
+        }
```

Comment:
It looks like it is possible to have Instant FX and a focused effect. Should that state be possible?